src/BizHawk.Client.Common/lua/CommonLibs/CommLuaLibrary.cs

```diff
@@ -133,43 +133,55 @@ private bool CheckSocketServer()
 		[LuaMethod("mmfSetFilename", "Sets the filename for the screenshots")]
 		public void MmfSetFilename(string filename)
 		{
-			CheckMmf();
 			APIs.Comm.MMF.Filename = filename;
 		}
 
 		[LuaMethod("mmfGetFilename", "Gets the filename for the screenshots")]
 		public string MmfGetFilename()
 		{
-			CheckMmf();
-			return APIs.Comm.MMF?.Filename;
+			return APIs.Comm.MMF.Filename;
 		}
 
 		[LuaMethod("mmfScreenshot", "Saves screenshot to memory mapped file")]
 		public int MmfScreenshot()
 		{
-			CheckMmf();
+			if (APIs.Comm.MMF.Filename == null)
+			{
+				Log("Memory mapped filename was not initialized, please initialize it via the command line");
+				return -1;
+			}
 			return APIs.Comm.MMF.ScreenShotToFile();
 		}
 
 		[LuaMethod("mmfWrite", "Writes a string to a memory mapped file")]
 		public int MmfWrite(string mmf_filename, string outputString)
 		{
-			CheckMmf();
 			return APIs.Comm.MMF.WriteToFile(mmf_filename, outputString);
 		}
+		[LuaMethod("mmfWriteBytes", "Write bytes to a memory mapped file")]
+		public int MmfWriteBytes(string mmf_filename, LuaTable byteArray)
+		{
+			return APIs.Comm.MMF.WriteToFile(mmf_filename, _th.EnumerateValues<double>(byteArray).Select(d => (byte)d).ToArray());
+		}
+		[LuaMethod("mmfCopyFromMemory", "Copy a section of the memory to a memory mapped file")]
+		public int MmfCopyFromMemory(string mmf_filename, long addr, int length, string domain)
+		{
+			return APIs.Comm.MMF.WriteToFile(mmf_filename, APIs.Memory.ReadByteRange(addr, length, domain).ToArray());
+		}
+		[LuaMethod("mmfCopyToMemory", "Copy a memory mapped file to a section of the memory")]
+		public void MmfCopyToMemory(string mmf_filename, long addr, int length, string domain)
+		{
+			APIs.Memory.WriteByteRange(addr, new List<byte>(APIs.Comm.MMF.ReadBytesFromFile(mmf_filename, length)), domain);
+		}
 		[LuaMethod("mmfRead", "Reads a string from a memory mapped file")]
 		public string MmfRead(string mmf_filename, int expectedSize)
 		{
-			CheckMmf();
-			return APIs.Comm.MMF?.ReadFromFile(mmf_filename, expectedSize);
+			return APIs.Comm.MMF.ReadFromFile(mmf_filename, expectedSize);
 		}
-
-		private void CheckMmf()

```

Comment:
This is hard to read, can you split the changeset into two commits? One to make `--mmf` optional, and one to add extra functions.

Reply:
Sure. I dont know if I can do that in this pull request, so maybe I will close this one and open a new one.

Reply:
You can force-push whatever set of commits you consider proper.